app/models/pbs/event/course.rb

```diff
@@ -42,6 +44,14 @@ module Pbs::Event::Course
     validates :bsv_days, numericality: { greater_than_or_equal_to: 0, allow_blank: true }
     validate :assert_bsv_days_precision
 
+    validates :number, format: { with: COURSE_NUMBER_REGEX,
+      message: :must_adhere_to_format,
+      if: :validate_number? }
+
+    def validate_number?
+      (new_record? || number_changed? || kind_id_changed?) && kind.validate_course_number
```

Comment:
```suggestion
      return false unless kind.validate_course_number?
      
      new_record? ||
        number_changed? ||
        kind_id_changed?     
```

Find ich persönlich leserlicher